presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -353,6 +353,7 @@ public void start()
             }
             catch (Throwable e) {
                 fail(e);
+                stateMachine.updateQueryInfo(Optional.empty());
```

Comment:
Does this look ok, @dain? An alternate change would be to not call fail, but do this instead:

    if (stateMachine.transitionToFailed(cause))
    {
        stateMachine.updateQueryInfo(Optional.empty());
    }

Another third possibility would be to change the return type of fail() to be a boolean and return what stateMachine.transitionToFailed(cause) returns.

WDYT is best?